setup.py

```diff
@@ -69,20 +71,38 @@ class CleanCommand(Clean):
 
     def run(self):
         Clean.run(self)
+        # Remove c files if we are not within a sdist package
+        cwd = os.path.abspath(os.path.dirname(__file__))
+        remove_c_files = not os.path.exists(os.path.join(cwd, 'PKG-INFO'))
+        if remove_c_files:
+            print('Clean will remove generate .c files')
         if os.path.exists('build'):
             shutil.rmtree('build')
         for dirpath, dirnames, filenames in os.walk('sklearn'):
             for filename in filenames:
-                if (filename.endswith('.so') or filename.endswith('.pyd')
-                        or filename.endswith('.dll')
-                        or filename.endswith('.pyc')):
+                if (filename.endswith('.so') or filename.endswith('.pyd') or
+                        filename.endswith('.dll') or
+                        filename.endswith('.pyc')):
                     os.unlink(os.path.join(dirpath, filename))
+                if remove_c_files:
+                    if filename.endswith('.c') and os.path.exists(
+                            os.path.join(dirpath,
+                                         filename[:-2] + '.pyx')):
+                        os.unlink(os.path.join(dirpath, filename))
+                    if filename.endswith('.cpp') and os.path.exists(
+                            os.path.join(dirpath,
+                                         filename[:-4] + '.pyx')):
+                        os.unlink(os.path.join(dirpath, filename))
             for dirname in dirnames:
                 if dirname == '__pycache__':
                     shutil.rmtree(os.path.join(dirpath, dirname))
+            if remove_c_files:
```

Comment:
however the next lines seem only to care about variables defined out of the loop.